.NET/Microsoft.Recognizers.Text.Number/Parsers/BaseIndianNumberParser.cs

```diff
@@ -0,0 +1,838 @@
+                    string sign = offset[0].Equals('-') ? string.Empty : "+";
+                    ret.Value = string.Concat(relativeTo, sign, offset);
+                    ret.ResolutionStr = GetResolutionStr(ret.Value);
+                }
+                else
+                {
+                    ret.Metadata.Offset = ret.ResolutionStr;
+
+                    // Every ordinal number is relative to the start
+                    ret.Metadata.RelativeTo = Constants.RELATIVE_START;
+                }
+            }
+
+            if (ret != null)
+            {
+                ret.Type = DetermineType(extResult);
+                ret.Text = ret.Text.ToLowerInvariant();
+            }
+
+            return ret;
+        }
+
+        protected ParseResult ParseTextNumber(ExtractResult extResult)
+        {
+            var result = new ParseResult
+            {
+                Start = extResult.Start,
+                Length = extResult.Length,
+                Text = extResult.Text,
+                Type = extResult.Type,
+                Metadata = extResult.Metadata,
+            };
+
+            var handle = extResult.Text;
+
+            handle = Config.HalfADozenRegex.Replace(handle, Config.HalfADozenText);
+
+            // Handling cases like "last", "next one", "previous one"
+            if ((this.Config.Config.Options & NumberOptions.SuppressExtendedTypes) == 0)
+            {
+                if (extResult.Metadata != null && extResult.Metadata.IsOrdinalRelative)
+                {
+                    return result;
+                }
+            }
+
+            var numGroup = handle.Split(Config.WrittenDecimalSeparatorTexts.ToArray(), StringSplitOptions.RemoveEmptyEntries);
+
+            var intPart = numGroup[0];
+            var stringMatch = TextNumberRegex.Match(intPart);
+
+            // Store all match str.
+            var matchStrs = new List<string>();
+
+            while (stringMatch.Success)
+            {
+                var matchStr = stringMatch.Groups[0].Value;
+                matchStrs.Add(matchStr);
+                stringMatch = stringMatch.NextMatch();
+            }
+
+            // Get the value recursively
+            var intPartRet = GetTheIntValue(matchStrs);
+
+            double pointPartRet = 0;
+            if (numGroup.Length == 2)
+            {
+                var pointPart = numGroup[1];
+                stringMatch = TextNumberRegex.Match(pointPart);
+                matchStrs.Clear();
+
+                while (stringMatch.Success)
+                {
+                    var matchStr = stringMatch.Groups[0].Value;
+                    matchStrs.Add(matchStr);
+                    stringMatch = stringMatch.NextMatch();
+                }
+
+                pointPartRet += GetPointValue(matchStrs);
+            }
+
+            result.Value = intPartRet + pointPartRet;
+
+            return result;
+        }
+
+        protected ParseResult ParseFracLikeNumber(ExtractResult extResult)
```

Comment:
This should override FracLikeNumberParse (and one of the two should be renamed for consistency). This new name is better.

Reply:
I also see many similarities between this method and FracLikeNumberParse. What are the key differences?